### DISCUSSION TITLE
syntax for serial device

### DISCUSSION THREAD

**Discussion by travisghansen** [cminyard/ser2net#135] [Q&A]
Hey, new to ser2net and I am completely lost about where I can find explanations about setting baud etc. I can see examples with options like `9600e71` and `115200N81` following the device name, but don't really see any doc about these strings. Can I get pointed to the correct place to understand the syntax and what the available options are? I can make some educated guesses for sure but I would love to read though the docs.

Thanks!

**Comment by cminyard** [comment]
On Mon, Dec 16, 2024 at 2:30 PM Travis Glenn Hansen <
***@***.***> wrote:

> Hey, new to ser2net and I am completely lost about where I can find
> explanations about setting baud etc. I can see examples with options like
> 9600e71 and 115200N81 following the device name, but don't really see any
> doc about these strings. Can I get pointed to the correct place to
> understand the syntax and what the available options are? I can make some
> educated guesses for sure but I would love to read though the docs.
>

The docs are in the gensio man page, try running "man 5 gensio".  Then
search for "serialdev".

ser2net really isn't just a serial to network connector any more, it's a
"gensio accepter" to "gensio connector" tool.  There are a ton of genios.

-corey

**Comment by travisghansen** [comment]
Thanks! That's very helpful. For this specific option it's less helpful but it exactly what I was after.

Any ideas where I can find more detailed info for the `speed` option?

```
This is a normal serial port configuration specification, like "9600N81".  The "speed=" at the beginning is optional, but "speed" is a default type for this.
```

Where would I discover what the `normal` port specs are? Sorry for the ignorance, new to this so the incoming knowledge that is likely generally assumed here cannot be assumed in my case :D

**Comment by cminyard** [comment]
On Tue, Dec 17, 2024 at 06:25:32AM -0800, Travis Glenn Hansen wrote:
> Thanks! That's very helpful. For this specific option it's less helpful but it exactly what I was after.
> 
> Any ideas where I can find more detailed info for the `speed` option?

That's the only place, unfortunately.

> 
> ```
> This is a normal serial port configuration specification, like "9600N81".  The "speed=" at the beginning is optional, but "speed" is a default type for this.
> ```
> 
> Where would I discover what the `normal` port specs are? Sorry for the ignorance, new to this so the incoming knowledge that is likely generally assumed here cannot be assumed in my case :D

The default is 9600N81.  I've added that.  The "default type" is
something else, you can set defaults for various things which let you
override the default value for pretty much anything.  It's part of the
gensio library, but you can set the in ser2net.  So if you wanted
the default to be 115200n81, you would do something like:
```
default:
   name: speed
   value: 115200n81
   class: serialdev
```
You can read about that in the ser2net.yaml.5 man page.

-corey

**Reply by travisghansen** [reply]
Sure, I understand that is the default and that I can change the default, but where can I discover what those lines *mean* and what the various options are? That seems to mean baud rate with some parity and data bits etc but I would like to see a comprehensive list of available options and their meaning.

**Comment by cminyard** [comment]
On Tue, Dec 17, 2024 at 07:10:03AM -0800, Travis Glenn Hansen wrote:
> Sure, I understand that is the default and that I can change the default, but where can I discover what those lines *mean* and what the various options are? That seems to mean baud rate with some parity and data bits etc but I would like to see a comprehensive list of available options and their meaning.

Ah.  That's kind of too big a discussion for the man page there.
I was kind of assuming people knew this stuff, but my son is a computer
science major and he doesn't know this stuff, so that's really a bad
assumption.

So I have added the specific values and a pointer:
```
The <speed> (or baud rate) setting is a number.  Depending on the
specific serial port, this can be almost any number.  Some serial port
on some systems support any arbatrary number, others have specific
numbers that are supported.

The <parity> setting is one of "N", "E", "O", "M", "S" for none, even,
odd, mark, and space.

The <databits> value is a number 5-8.  It sets the number of bits sent
for each character, and is generally 8.

The <stopbits> values is the number of bits to mark the end of a
character, either 1 or 2.

Really just need to know what the value you are connecting to is.  For
more details, see https://en.wikipedia.org/wiki/Serial_port.
```

**Reply by travisghansen** [reply]
Perfect, I was after an understanding of what each position was and what the possible values are. That answered all of that!

**Reply by travisghansen** [reply]
And now I see I completely missed the syntax called out after the speed= in the doc, sorry for blindness.
